RecoLocalCalo/EcalDeadChannelRecoveryAlgos/src/EcalDeadChannelRecoveryBDTG.cc

```diff
@@ -0,0 +1,239 @@
+//  Original Authors:  S. Taroni, N. Marinelli
+//  University of Notre Dame - US
+//  Created:   
+//
+//
+//
+
+
+#include "RecoLocalCalo/EcalDeadChannelRecoveryAlgos/interface/EcalDeadChannelRecoveryBDTG.h"
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h" // can I use a egammatools here?
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include <iostream>
+#include <TMath.h>
+#include <TDirectory.h>
+
+
+#include<iostream>
+#include<ostream>
+#include<string>
+#include<fstream>
+
+template <>
+void EcalDeadChannelRecoveryBDTG<EBDetId>::loadFile() {
+
+  readerNoCrack = new TMVA::Reader( "!Color:!Silent" );
+
+  readerNoCrack->AddVariable("E1/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx.rEn[0]));
+  readerNoCrack->AddVariable("E2/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx.rEn[1]));
+  readerNoCrack->AddVariable("E3/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx.rEn[2]));
+  readerNoCrack->AddVariable("E4/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx.rEn[3]));
+  readerNoCrack->AddVariable("E6/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx.rEn[5]));
+  readerNoCrack->AddVariable("E7/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx.rEn[6]));
+  readerNoCrack->AddVariable("E8/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx.rEn[7]));
+  readerNoCrack->AddVariable("E9/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx.rEn[8]));
```

Comment:
I looked at this when copy-pasting come code for my GBRForest test, and I just want to point out that in case of the loop a `if(i == 4) continue;` would be needed, because there is no `E5/...` variable. Also I noticed one can just do `"E" + std::to_string(i+1) + "/(E1+E2+E3+E4+E6+E7+E8+E9)"` and therefore avoid some ROOT class.

Reply:
(AddVariable argument type is TString)
indeed, to_string should work.
Thanks!

Reply:
rolled up for the next commit